docs/source/QuickStart.rst

```diff
@@ -0,0 +1,163 @@
+In this directory, you will see three files: 
+
+* **echo.py**: This is our actual function logic, it can be anything you want as
+  long as all of it's dependencies are in the echo/ folder that we pass to
+  SRK. 
+* **f.py**: This is the open-lambda glue code. OL requires that functions be
+  named ``f()`` and live in ``f.py``. The signature of ``f()`` must be
+  preserved. The body of ``f()`` does any OL-specific actions and calls
+  ``echo()`` with the correct arguments.
+* **aws.py**: This is the aws Lambda glue code. AWS lambda is more flexible with
+  function naming, but has an incompatible signature requirement (it includes a
+  'context' field that OL doesn't). SRK requires that you include an ``aws.py``
+  with a function ``f(event, context)``.
+
+Since ``f.py`` and ``aws.py`` provide only simple wrappers, we will focus on
+``echo.py``. This file provides our cloud function ``echo()`` which simply
+returns any request it receives.
+
+*************************
+SRK Configuration
+*************************
+SRK provides an interchangable interface to multiple providers of standard
+cloud services. In this case, we will focus on two different FaaS providers:
+AWS Lambda and OpenLambda. At the moment, SRK does not manage the configuration
+of these providers, so we'll need to do that now. First, let's create a default
+configuration from the template provided with SRK:
+
+::
+
+   $ cp configs/example-srk.yaml configs/srk.yaml
+
+In the next two subsections, we will customize this config to include
+OpenLambda and AWS Lambda support.
+
+OpenLambda
+=================
+OpenLambda is an open-source function-as-a-service provider based Docker. We
+maintain a fork of this project to provide additional features. Go ahead and
+clone this repo anywhere in your filesystem, we'll use our home directory in
+these instructions:
+
+::
+
+   $ cd ~
+   $ git clone git@github.com:NathanTP/open-lambda.git
+
+Next, follow the instructions in open-lambda's README to compile and install
+the system. You will need to have Docker and Golang installed before starting
+this step.
+
+With OpenLambda built, we can now configure SRK to use it. Open up
+``configs/srk.yaml`` and modify the ``service.faas.openLambda`` section to look
+as follows:
+
+::
+
+   faas:
+      openLambda:
+         olcmd: "~/open-lambda/ol"
+         oldir: "~/open-lambda/default-ol"
+
+AWS Lambda
+=============
+A full guide to using AWS Lambda is beyond the scope of this document, but you
+can follow AWS's tutorial `here
+<https://docs.aws.amazon.com/lambda/latest/dg/getting-started.html>`_. For SRK
+to work, you will need to provide an ARN role and optional VPC. You can set
+these values in ``configs/srk.yaml`` in the ``service.faas.awsLambda`` section.
+For example:
+
+::
+
+   faas:
+      awsLambda:
+         role: "arn:aws:iam::123459789012:role/service-role/my-service-role-ae04d032"
+         vpc: "vpc-123456789abcdef,sg-123456789abcdef"
```

Comment:
How does the security group need to be configured?

Reply:
no idea. I added some more stuff here. The VPC can really just be NULL unless you need something special so I changed it to default to that. Again, anything you can add to the docs here would be great.